modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Foldable.kt

```diff
@@ -71,6 +71,44 @@ interface Foldable<F> {
       }
     }
 
+  /**
+   * Transforms this [Foldable] into a new one which doesn't contain the given [element].
+   * Will return an identical [Foldable] if it doesn't contain [element] in the first place.
+   * Example:
+   *
+   * {: data-executable='true'}
+   *
+   * ```kotlin:ank
+   * import arrow.core.ListK
+   * import arrow.core.extensions.listk.applicative.applicative
+   * import arrow.core.extensions.listk.foldable.foldable
+   * import arrow.core.extensions.listk.monoidK.monoidK
+   *
+   * fun main(args: Array<String>) {
+   *   //sampleStart
+   *   ListK.foldable().run { listOf(1, 2, 3, 4).k().remove(2, ListK.monoidK(), ListK.applicative()) }
+   *  //sampleEnd
+   *  println(result)
+   * }
+   * ```
+   *
+   * @param element the element to remove
+   * @param monoidK the [Monoid] which produces the empty result [Foldable]
+   * @param AF the [Applicative] which will be used to lift the values in this [Foldable]
+   */
+  fun <A> Kind<F, A>.remove(
```

Comment:
I am not sure if this method makes sense at all. I checked in with Haskell and Cat's Library and none of them implement a remove function. 
Nonetheless, this remove function is isomorphic to a simple fold function, where you preserve the Datatype which implements Foldable and just exclude the element you don't want. This makes more sense if you keep it in the Datatype as a function for convenience, but I object having it in this type class.

Reply:
I do not object on having it here and I'd like to see it in `FunctorFilter` too with a simpler signature.

Just wondering if there is a type that has an instance for `Foldable` and `MonoidK` but no instance for `FunctorFilter` 🤔 cc\ @raulraja @pakoito

Reply:
I am not saying that we should not have a remove function, but it should not be a typeclass-level function. We can have them for datatypes and refer to them with a fold. 
I don't see the point of a simpler API, if we add this function.

Reply:
@i-walker please discuss this in the relevant place where there is a very long conversation which also explains why this is here: [link](https://kotlinlang.slack.com/archives/C8UK6RTHU/p1563102430078300).